core/src/qrcode/QRBitMatrixParser.cpp

```diff
@@ -50,8 +50,21 @@ const Version* ReadVersion(const BitMatrix& bitMatrix, bool isMicro)
 	int dimension = bitMatrix.height();
 
 	int provisionalVersion = (dimension - Version::DimensionOffset(isMicro)) / Version::DimensionStep(isMicro);
+	if (isMicro) {
+		int verticalVersionBits = 0;
+		for (int y = 8; y < dimension; ++y)
+			AppendBit(verticalVersionBits, getBit(bitMatrix, 0, y, false));
+		int horizontalVersionBits = 0;
+		for (int x = 8; x < dimension; ++x)
+			AppendBit(horizontalVersionBits, getBit(bitMatrix, x, 0, false));
+		auto theParsedVersion = Version::DecodeVersionInformation(verticalVersionBits, horizontalVersionBits);
+		if (theParsedVersion != nullptr && theParsedVersion->dimensionForVersion() == dimension)
+			return theParsedVersion;
+		return nullptr;
+	}
+
```

Comment:
Forgive me if I'm wrong but I don't see how this change improves the detection rate in any way. It actually decreases it: you simply check the integrity of the two timing patterns and fail the decoding if just any one of those bits is flipped. This basically reduces the error tolerance of the symbol to 0% if the error occurs inside the timing pattern. What was your intention for this change, if I may ask?

The version of a MicroQRCode is in fact encoded inside the format info and that is where I intend to parse it from in my new detector implementation. (This whole concept of strictly separating detection and decoding of the symbol is ill advised here. See also the suggested 'reference algorithm' in the specification.)

Reply:
I see what you mean. It still does not make sense to me, neither does the reference algo, to be fair. They are not doing the same thing though: The spec algo differs in that it does only enforce the number of dark modules of both timing patterns to be equal and between 2 and 5 (only to ignore that info later on anyway). Your implementation requires them to be both exactly as anticipated by the version number estimate determined by other means. Anyway, sorry if this sounds like dogmatism.

Reply:
Update: now that I tried your code for the first time after I've got my first results from my new detector, I noticed that it is plain wrong: you sampled the wrong bits and use the masks for QR instead of MQR codes. So this code will indeed go away. ;).